src/Controllers/HistoryViewerController.php

```diff
@@ -339,4 +419,31 @@ protected function scaffoldForm($formName, array $context = [], array $extra = [
             LeftAndMainFormRequestHandler::create($form, $extra)
         );
     }
+
+    private function getPostedJsonValue(HTTPRequest $request, string $key): mixed
+    {
+        $data = json_decode($request->getBody(), true);
+        if (!array_key_exists($key, $data)) {
+            $this->jsonError(400);
+        }
+        return $data[$key];
+    }
```

Comment:
You added something very similar (if not identical) to this in `AssetAdmin` - is there any sense in centralising this logic somewhere? Perhaps even on the request itself?

Reply:
It's such a small amount of code I think the centralisation probably just increases complexity rather than reduces it

Reply:
Looks like you ended up moving it into `LeftAndMain` 👍